src/main/java/dev/openfeature/sdk/providers/memory/InMemoryProvider.java

```diff
@@ -60,14 +60,16 @@ public void initialize(EvaluationContext evaluationContext) throws Exception {
     }
 
     /**
-     * Updating provider flags configuration, replacing existing flags.
-     * @param flags the flags to use instead of the previous flags.
+     * Updates the provider flags configuration.
+     * For existing flags, the new configurations replace the old one.
+     * For new flags, they are added to the configuration.
+     * @param newFlags the new flag configurations
      */
-    public void updateFlags(Map<String, Flag<?>> flags) {
-        Set<String> flagsChanged = new HashSet<>();
-        flagsChanged.addAll(this.flags.keySet());
-        flagsChanged.addAll(flags.keySet());
-        this.flags = new HashMap<>(flags);
+    public void updateFlags(Map<String, Flag<?>> newFlags) {
+        Set<String> flagsChanged = new HashSet<>(newFlags.keySet());
+        flagsChanged.removeAll(this.flags.keySet());
```

Comment:
Oh, nice catch.

Ya I think it's fine just to list all flags here. This is just a testing util. 

@ryanprayogo you can mention that all flags will be put into the change notification. In a way that's technically correct anyway since the set is immutable. This is a testing util primarily, and anyway providers get to decide what constitutes "changed" for their purposes.